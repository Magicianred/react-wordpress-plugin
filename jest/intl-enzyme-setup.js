/* eslint-disable import/no-extraneous-dependencies */
/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 *
 * Docs for the code below
 *  - https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#enzyme
 *  - https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#snapshot-testing
 */
import React from 'react'
import { IntlProvider, intlShape } from 'react-intl'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'

// You can pass your messages to the IntlProvider. Optional: remove if unneeded.
// Create the IntlProvider to retrieve context for wrapping around.
// NOTE: currently before each test the we have to build the app to get this file!
import messages from '../dist/static/languages/en.json'

const intlProvider = new IntlProvider({ locale: 'en', messages }, {})
const { intl } = intlProvider.getChildContext()

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl })
}

export function shallowWithIntl(node, { context } = {}) {
  return shallow(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
  })
}

export function mountWithIntl(node, { context, childContextTypes } = {}) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes),
  })
}

export const createComponentWithIntl = (children, props = { locale: 'en' }) =>
  renderer.create(
    <IntlProvider {...props} locale="en" messages={messages}>
      {children}
    </IntlProvider>,
  )

// as we also use PropTypes we need to mock the Intl PropTypes
export const mockIntlShape = {
  formatDate: jest.fn(),
  formatTime: jest.fn(),
  formatRelative: jest.fn(),
  formatNumber: jest.fn(),
  formatPlural: jest.fn(),
  formatMessage: jest.fn(),
  formatHTMLMessage: jest.fn(),
  now: jest.fn(),
}
