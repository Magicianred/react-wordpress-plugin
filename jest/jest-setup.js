/* eslint-disable import/no-extraneous-dependencies */
const Enzyme = require('enzyme')
const Adapter = require('enzyme-adapter-react-16')

Enzyme.configure({ adapter: new Adapter() })

// usually injected by WordPress (see wp_localize_script in class-admin.php)
global.wpGlobals = {
  namespace: 'plugin-name/v1',
  nonce: 'example-nonce-123',
  root: 'https://example.com/wp-json/',
}

// helper to wait for something (eg. async actions)
global.nextTick = () => new Promise(res => process.nextTick(res)) // eslint-disable-line
