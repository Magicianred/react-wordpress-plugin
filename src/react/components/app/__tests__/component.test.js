import React from 'react'
import { createComponentWithIntl } from '../../../../../jest/intl-enzyme-setup'
import App from '../component'

// useful tipps for testing with JEST
// - https://marmelab.com/blog/2015/06/24/jest-in-practice.html
describe('Components/App', () => {
  it('renders app component', () => {
    const wrapper = createComponentWithIntl(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
