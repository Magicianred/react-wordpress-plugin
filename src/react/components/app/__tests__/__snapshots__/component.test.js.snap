// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/App renders app component 1`] = `
<div
  className="app-plugin-name"
>
  <div
    className="app-plugin-name__container"
  >
    <WithStyles(PersistentDrawer)
      content={
        Array [
          Object {
            "component": [Function],
            "icon": [Function],
            "name": "Manage",
            "props": Object {},
          },
          Object {
            "component": [Function],
            "icon": [Function],
            "name": "Settings",
            "props": Object {},
          },
        ]
      }
    />
  </div>
</div>
`;
